src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/WriteYamlStep.java

```diff
@@ -139,6 +146,24 @@ private boolean isValidObjectType(Object obj) {
 
     @Override
     public StepExecution start(StepContext context) throws Exception {
+        if (this.returnText) {
+            if (this.file != null) {
+                throw new IllegalArgumentException("cannot provide both returnText and file to writeYaml");
```

Comment:
Why not? Cound't you simply just check if `returnText` is set at the end of the current execution and if so return the text otherwise `null`?

Reply:
I'm not sure what you mean. If we ask SnakeYaml to output to a file then we don't have a string to return.

Reply:
Could output to a ByteArrayOutputStream, but that is overkill for now :)